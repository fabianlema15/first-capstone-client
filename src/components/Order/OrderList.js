import React from 'react';
import GenericContext from '../../contexts/GenericContext'
import OrderProductItem from './OrderProductItem';
import OrderPromotionItem from './OrderPromotionItem';
import Helper from '../Utils/Helper'

class OrderList extends React.Component{
  static contextType = GenericContext;

  renderObjs() {
    let objList = [];
    if (!this.props.objList){
      objList = this.context.objList
      return Object.keys(objList).map(key =>
        <OrderProductItem key={objList[key].id} obj={objList[key]} objLabel={Helper.serializeObj(this.context.getObjArrayProd)}/>
      )
    }else{
      objList = this.props.objList
      return Object.keys(objList).map(key =>
        <OrderPromotionItem key={objList[key].id} obj={objList[key]} objLabel={Helper.serializeObj(this.context.getObjArrayProm)}/>
      )
    }

  }

  render(){
    return <div>
            {this.renderObjs()}
    </div>
  }
}

export default OrderList;
