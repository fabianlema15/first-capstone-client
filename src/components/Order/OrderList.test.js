import React from 'react';
import ReactDOM from 'react-dom';
import OrderList from './OrderList'
import toJson from 'enzyme-to-json'
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe('<OrderList />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <OrderList objList={{}}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Renders OrderList as expected', () => {
    const context = { objName: 'Test', objList: {}};
	  const wrapper = shallow(<OrderList objList={{}}/>, { context });
	  expect(toJson(wrapper)).toMatchSnapshot();
	  });

})
