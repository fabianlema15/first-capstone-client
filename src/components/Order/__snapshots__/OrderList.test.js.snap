// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderList /> Renders OrderList as expected 1`] = `<div />`;
